### ISSUE TITLE
Search Clicking and Quick Switch to Gallery View

### ISSUE THREAD

**Issue by Horuyuki** [Taiko2k/TauonMusicBox#508]
Hello kind dev!

I was fiddling around with the app today and had 2 things to suggest, would be very nice if you can implement them!

1. I have set the selection of tracks to double click, yet when in search, it still plays it upon single clicking the track.
Reason: The sudden change of tracks sometimes gives a spike in volume because different mixes, I would like to be able to browse the search results by single clicking on it first before it is gonna start playing.

2. A hot key for showcase/lyrics lay out mode with a quick keyboard short cut?
Reason: I don't have a 5 key mouse, just a normal one so that is a no go. Would you be able to either add a defined hot key switching or adding a short cut custom definer?

Thanks for the nice player!

Version: 6.5.4
AUR installation.

**Comment by Taiko2k** [comment]
Kind dev? Ha, I'm a mean dev! :smiling_imp: 

> I would like to be able to browse the search results by single clicking on it first before it is gonna start playing.

The intended way of achieving this is to right click the result which will take you to the result in the tracklist.

> A hot key for showcase/lyrics lay out mode with a quick keyboard short cut?

By default, showcase/lyrics layout can be toggled using <kbd>F5</kbd>. You can rebind that and others by going MENU > Settings > Open Data Folder, then editing `input.txt`.

**Comment by Horuyuki** [comment]
> Kind dev? Ha, I'm a mean dev! smiling_imp
> 
> > I would like to be able to browse the search results by single clicking on it first before it is gonna start playing.
> 
> The intended way of achieving this is to right click the result which will take you to the result in the tracklist.
> 
> > A hot key for showcase/lyrics lay out mode with a quick keyboard short cut?
> 
> By default, showcase/lyrics layout can be toggled using F5. You can rebind that and others by going MENU > Settings > Open Data Folder, then editing `input.txt`.

Wow that was really quick and throughout! So I have some more, questions would be a better fit it seems:

> The intended way of achieving this is to right click the result which will take you to the result in the tracklist.
Would you mind making it highlight the album cover if in gallery mode too?

> By default, showcase/lyrics layout can be toggled using F5. You can rebind that and others by going MENU > Settings > Open Data Folder, then editing input.txt.
That worked beautifully!

1. (this might be a bug/improvement) So I am in this audio circuit: `Gstreamer --> ALSA --> Gnome/Arch linux output device chosen as USB DAC; local files - flac` And even though the fade option is enabled, it is still a bit short. Maybe add a longer/customizable fade time? Or could it be from other problems?

2. There are no audio outputting natively form the app when streaming to discord, sharing music with my friends. I have to turn on the mic which then has a filters applied by disocrd. Configuration on my side or fixing needed on dev side?
I did enable the rich presence though it doesn't seem like discord is picking it up? Is this related?

**Comment by Taiko2k** [comment]
> And even though the fade option is enabled, it is still a bit short.

Er um mumble <sub>mumble <sub>mumble...</sub></sub> 

So... in regards to the GStreamer backend, In the Tauon config file theres a setting to change the fade time, but now that I look, I had disabled it and it has no effect currently. Maybe I should re-enable it next release. (This is pause/stop only btw, there isn't jump fade with the GStreamer backend) Though regardless fades with GStreamer are kind of crap.

> There are no audio outputting natively form the app when streaming to discord, sharing music with my friends. I have to turn on the mic which then has a filters applied by disocrd. Configuration on my side or fixing needed on dev side?

I'm not really sure. By default mixing apps with pure ALSA does require extra config. Its a pain which is why PulseAudio/Pipewire exist.

**Comment by Horuyuki** [comment]
> I had disabled it and it has no effect currently. Maybe I should re-enable it next release. (This is pause/stop only btw, there isn't jump fade with the GStreamer backend) Though regardless fades with GStreamer are kind of crap.
I am perfectly fine living off with only pause play, as some tracks I want it to start a full volume anyways.
But if I am following correctly, that means even if I enable the fade in config with a longer time, it will still be of no use? Or did you mean because you disabled it leading there to be no effect?


> I'm not really sure. By default mixing apps with pure ALSA does require extra config. Its a pain which is why PulseAudio/Pipewire exist.
So I should try the following?
1. Gstream to pulse and see if discord streams native sound
2. Use the PHAzOR and USB DAC option and see if discord streams native sound

And discord doesn't show me using the tauon box, is there a way to see your own rich presence instead of going to another account?

**Comment by Taiko2k** [comment]
> And discord doesn't show me using the tauon box, is there a way to see your own rich presence instead of going to another account?

You should be able to see it if you right click your name in the chat.

> So I should try the following

I'm not really sure, it sounds like a kind of an advanced setup since you want to route audio to two locations. Does this work for you with other apps? I'm sure its possible if you know what you're doing.

If it were me I'd try Pipewire + Cadence, but that's kind of advanced. Sorry I can't really offer help with this.

**Comment by Horuyuki** [comment]
I have no idea how to make it work with the combination pipewire+cadence. But that is no big deal, I will just the mic input of discord, even though the sound will be forced mono and compressed to smth like MP3, it would suffice. I just wanted to note how streaming with music bee in windows will be able to output audio in full spec. Just putting it out there.

Lastly, thanks for the great player with such a nice UI!

I was thinking that you guys can let us customize themes, but I found a pretty good default one. The folder hierarchy that is specific to the files imported was so very useful. I was trying to find that kinda thing in music bee but it defaults to system wide folder browsing instead. Amazing job you guys!

I just installed the phone remote, it was superb and very useful.

One little suggestion as a closing word: Maybe add a button which can instant update a user defined music source folder instead of stuff from Downloads only? Since then if I move it somewhere else it will be gone from the library. If I can organize the specified folder space and "rescan" that spot specifically I can see my results immediately.

Or I am just dumb and there is already this option, I just don't know where it is.
